cpp/ql/src/experimental/Security/CWE/CWE-020/LateCheckOfFunctionArgument.ql

```diff
@@ -0,0 +1,66 @@
+/**
+ * @name Late Check Of Function Argument
+ * @description --Checking the function argument after calling the function itself.
+ *              --This situation looks suspicious and requires the attention of the developer.
+ *              --It may be necessary to add validation before calling the function.
+ * @kind problem
+ * @id cpp/late-check-of-function-argument
+ * @problem.severity warning
+ * @precision medium
+ * @tags correctness
+ *       security
+ *       external/cwe/cwe-20
+ */
+
+import cpp
+import semmle.code.cpp.valuenumbering.GlobalValueNumbering
+
+predicate numberArgument(Function f, int size) {
+  f.hasGlobalOrStdName("write") and size = 2
+  or
+  f.hasGlobalOrStdName("read") and size = 2
+  or
+  f.hasGlobalOrStdName("lseek") and size = 1
+  or
+  f.hasGlobalOrStdName("memmove") and size = 2
+  or
+  f.hasGlobalOrStdName("memset") and size = 2
+  or
+  f.hasGlobalOrStdName("memcpy") and size = 2
+  or
+  f.hasGlobalOrStdName("memcmp") and size = 2
+  or
+  f.hasGlobalOrStdName("strncat") and size = 2
+  or
+  f.hasGlobalOrStdName("strncpy") and size = 2
+  or
+  f.hasGlobalOrStdName("strncmp") and size = 2
+  or
+  f.hasGlobalOrStdName("snprintf") and size = 1
+  or
+  f.hasGlobalOrStdName("strndup") and size = 2
+  or
+  f.hasGlobalOrStdName("read") and size = 2
+}
+
+class IfCompareWithZero extends IfStmt {
+  IfCompareWithZero() { this.getCondition().(RelationalOperation).getAChild().getValue() = "0" }
+
+  Expr noZerroOperand() {
+    if this.getCondition().(RelationalOperation).getGreaterOperand().getValue() = "0"
+    then result = this.getCondition().(RelationalOperation).getLesserOperand()
+    else result = this.getCondition().(RelationalOperation).getGreaterOperand()
+  }
+}
+
+from FunctionCall fc, IfCompareWithZero ifc, int na
+where
+  numberArgument(fc.getTarget(), na) and
+  na >= 0 and
```

Comment:
I don't think you need this line.  `na` is constrained to the values provided in `numberArgument`.